updater/spec/dependabot/updater/dependency_group_change_batch_spec.rb

```diff
@@ -54,5 +54,104 @@
+      end
+
+      after do
+        Dependabot::Experiments.reset!
+      end
+
+      let(:dependency_change) do
+        Dependabot::DependencyChange.new(
+          job: job,
+          updated_dependency_files: updated_dependency_files,
+          updated_dependencies: updated_dependencies,
+          dependency_group: group
+        )
+      end
+
+      let(:updated_dependency_files) do
+        [
+          Dependabot::DependencyFile.new(
+            name: "Gemfile", content: "mock-updated-gemfile", directory: "/"
+          ),
+          Dependabot::DependencyFile.new(
+            name: "Gemfile.lock", content: "mock-updated-gemfile-lock", directory: "/hello/.."
+          ),
+          Dependabot::DependencyFile.new(
+            name: "Gemfile", content: "mock-updated-package-json", directory: "/elsewhere"
+          ),
+          Dependabot::DependencyFile.new(
+            name: "Gemfile", content: "mock-package-json", directory: "unknown"
+          ),
+          Dependabot::DependencyFile.new(
+            name: "Gemfile", content: "mock-package-json", directory: "../../oob"
+          )
+        ]
+      end
+
+      let(:updated_dependencies) do
+        [
+          Dependabot::Dependency.new(
+            name: "test-dependency-1",
+            package_manager: "bundler",
+            version: "1.1.0",
+            requirements: [
+              {
+                file: "Gemfile",
+                requirement: "~> 1.1.0",
+                groups: ["test"],
+                source: nil
+              }
+            ],
+            directory: "/"
+          ),
+          Dependabot::Dependency.new(
+            name: "test-dependency-2",
+            package_manager: "bundler",
+            version: "1.1.0",
+            requirements: [
+              {
+                file: "Gemfile",
+                requirement: "~> 1.1.0",
+                groups: ["test"],
+                source: nil
+              }
+            ],
+            directory: "/hello/.."
+          )
+        ]
+      end
+
+      let(:group) do
+        Dependabot::DependencyGroup.new(
+          name: "test",
+          rules: { patterns: ["test-*"] }
+        )
+      end
+
+      it "still tracks the updated_dependencies in the global list" do
+        batch = described_class.new(initial_dependency_files: files)
+        batch.merge(dependency_change)
+
+        expect(batch.updated_dependencies).to eq(updated_dependencies)
+      end
+
+      it "tracks the updated dependencies in the appropriate updated_dependency_files" do
+        batch = described_class.new(initial_dependency_files: files)
+        batch.merge(dependency_change)
+
+        # FIXME: All of the files in the dependency_file_batch will contain a copy of the updated dependencies.
+        # Eventually this should be narrowed down so that only the files that were modified
+        # by the newly merged dependency_change have their updated_dependencies updated.
```

Comment:
🤔 What will the impact of this be in the short term?

Reply:
This field only exists in-memory so it's essentially thrown away when the DependencyGroupChangeBatch is dereferenced.

This PR sets the field up so it can be used in a follow-up PR.